hack/sshtool/azure.go

```diff
@@ -0,0 +1,169 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	mgmtnetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-07-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+// disable removes load balancer and master nic configuration associated with
+// SSH
+func (s *sshTool) disable(ctx context.Context) error {
+	for i := 0; i < 3; i++ {
+		nicName := fmt.Sprintf("%s-master%d-nic", s.infraID, i)
+
+		nic, err := s.interfaces.Get(ctx, s.clusterResourceGroup, nicName, "")
+		if err != nil {
+			return err
+		}
+
+		disableNIC(&nic)
+
+		s.log.Printf("updating %s", nicName)
+		err = s.interfaces.CreateOrUpdateAndWait(ctx, s.clusterResourceGroup, nicName, nic)
+		if err != nil {
+			return err
+		}
+	}
+
+	lbName := s.infraID + "-internal-lb"
+
+	lb, err := s.loadBalancers.Get(ctx, s.clusterResourceGroup, lbName, "")
+	if err != nil {
+		return err
+	}
+
+	disableLB(&lb)
+
+	s.log.Printf("updating %s", lbName)
+	return s.loadBalancers.CreateOrUpdateAndWait(ctx, s.clusterResourceGroup, lbName, lb)
+}
+
+// enable adds load balancer and master nic configuration associated with SSH
+func (s *sshTool) enable(ctx context.Context) error {
+	lbName := s.infraID + "-internal-lb"
+
+	lb, err := s.loadBalancers.Get(ctx, s.clusterResourceGroup, lbName, "")
+	if err != nil {
+		return err
+	}
+
+	enableLB(&lb)
+
+	s.log.Printf("updating %s", lbName)
+	err = s.loadBalancers.CreateOrUpdateAndWait(ctx, s.clusterResourceGroup, lbName, lb)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < 3; i++ {
+		nicName := fmt.Sprintf("%s-master%d-nic", s.infraID, i)
+
+		nic, err := s.interfaces.Get(ctx, s.clusterResourceGroup, nicName, "")
+		if err != nil {
+			return err
+		}
+
+		enableNIC(&nic, &lb, i)
+
+		s.log.Printf("updating %s", nicName)
+		err = s.interfaces.CreateOrUpdateAndWait(ctx, s.clusterResourceGroup, nicName, nic)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func disableNIC(nic *mgmtnetwork.Interface) {
+	backendAddressPools := make([]mgmtnetwork.BackendAddressPool, 0, len(*(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools))
+	for _, p := range *(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools {
+		if p.Name != nil && !strings.HasPrefix(*p.Name, "ssh-") {
+			backendAddressPools = append(backendAddressPools, p)
+		}
+	}
+
+	(*nic.IPConfigurations)[0].LoadBalancerBackendAddressPools = &backendAddressPools
+}
+
+func enableNIC(nic *mgmtnetwork.Interface, lb *mgmtnetwork.LoadBalancer, i int) {
+	disableNIC(nic)
```

Comment:
Why this is here?

Reply:
if it weren't there and enableNIC were called twice, it would try to append the same pool twice, which would be invalid